### PR TITLE
[WIP] Implement JSON Generator to write Output from Parser.

### PR DESCRIPTION
**Pull request by SurajAralihalli**
Closes [#1831 ](https://github.com/NVIDIA/spark-rapids-jni/issues/1831)
[WIP] Pending - testcases, write_float(), code cleanup

### CODE REVIEW

File: src/main/cpp/src/get_json_object.hpp [hunk 1]
```diff
@@ -23,6 +23,265 @@
...
+  {
+    if (get_output_start_position() != nullptr) {
+      return std::string(get_output_start_position(), get_output_len());
+    } else {
+      return "";
+    }
+  }
+
+ private:
+  bool has_members[max_json_nesting_depth] = {false};
```
Comment (res-life) [inline src/main/cpp/src/get_json_object.hpp:235]: json_parser is already scanning the json tokens and comma/colon chars.
json_parser also maintains the nesting states.
We can put the writing logic into json_parser when parsing.

File: src/main/cpp/tests/CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +63,9 @@ ConfigureTest(DATETIME_REBASE
 ConfigureTest(ROW_CONVERSION
     row_conversion.cpp)
 
+ConfigureTest(JSON_GENERATOR
+    json_generator_tests.cpp)
```
Comment (res-life) [inline src/main/cpp/tests/CMakeLists.txt:67]: Nit, put cases into GET_JSON_OBJECT binary.

File: src/main/cpp/src/get_json_object.hpp [hunk 3]
```diff
@@ -23,6 +23,265 @@
...
+    if (nullptr == output) {
+      return json_generator();
+    } else {
+      return json_generator(output + output_len, 0);
+    }
+  }
+
+  CUDF_HOST_DEVICE json_generator finish_child_generator(json_generator const& child_generator)
+  {
+    // logically delete child generator
```
Comment (ttnghia) [inline src/main/cpp/src/get_json_object.hpp:53]: What does this mean? We do that at the caller?

Comment (SurajAralihalli) [reply src/main/cpp/src/get_json_object.hpp:53]: Thanks Nghia, I think this PR is out of scope now as parser maintains context and implements `copy_current_structure`. I'll have this PR taken down when the PR #1868 is merged. 

Are we using this function `finish_child_generator` in the new implementation? @res-life

Comment (res-life) [reply src/main/cpp/src/get_json_object.hpp:53]: >Are we using this function finish_child_generator in the new implementation? @res-life

No.

**Pull request closed without merging**